### DISCUSSION TITLE
Getting weird distortion and a corrupted image when converting from PNG to TIF.

### DISCUSSION THREAD

**Discussion by futuremotiondev** [ImageMagick/ImageMagick#5575] [Help] [answered]
As my post title says, converting a few images from PNG to TIF, and then back to PNG is resulting with a corrupted image. 

Here is a screen recording:

https://user-images.githubusercontent.com/33441569/190929332-7eb7e7dc-29b6-43f3-8e10-bbff0daca438.mp4

Here is a zip of PNG files I'm testing with:

[PNG Testing.zip](https://github.com/ImageMagick/ImageMagick/files/9595306/PNG.Testing.zip)

Does anyone know why this is happening and how to fix it?

Any help is greatly appreciated.

Update:
It's not just PNG. Converting the TIF to JPG and BMP also result with a corrupted image.

**Comment by fmw42** [comment]
What is your command sequence?

**Reply by fmw42** [reply]
Please post text of your command sequence not a video. The video is too hard to read

**Reply by fmw42** [reply]
What is your Photoshop version?

**Reply by fmw42** [reply]
What is your IM version?

**Reply by futuremotiondev** [reply]
I literally show it in the screen recording. 

`magick "Actions Buy32x32.png" result.tiff`
and then:
`magick result.tif result.png`

**Reply by futuremotiondev** [reply]
Photoshop version is CS5.1, but I also have Photoshop 2022. I prefer to use CS5 for most simple tasks since it's faster. 

ImageMagick version is: `Version: ImageMagick 7.1.0-47 Q16-HDRI x64 15861e0:20220827 https://imagemagick.org`

Commands are:
`magick "Actions Buy32x32.png" result.tiff`
and then:
`magick result.tif result.png`

The image is corrupted in Windows explorer as well, so it's not a Photoshop compatibility problem.

Try the command sequence on the files that I uploaded. See if the results are corrupted for you.

**Reply by fmw42** [reply]
Works fine for me on IM 7.1.0-48 Mac OSX Monterey with PS 23.51

try

```
magick "Actions Buy32x32.png" -type truecoloralpha result.tiff

magick result.tif PNG32:result.png
```

**Reply by futuremotiondev** [reply]
Didn't work. Still created a corrupted image.
 
![result](https://user-images.githubusercontent.com/33441569/190929914-127b7c38-37e3-4d76-8367-a6136c7ea278.png)

**Comment by futuremotiondev** [comment, marked as answer]
Solved it! It was a bug with the previous version of IM. I just updated to the latest and the corruption no longer occurs.

**Reply by fmw42** [reply]
See https://github.com/ImageMagick/Website/blob/main/ChangeLog.md

png.c: Remove the extra space character in "%8lu " https://github.com/ImageMagick/ImageMagick/pull/5513
